src/ParamHandler.ts

```diff
@@ -104,9 +106,19 @@ export class ParamHandler {
     private parseValue(value: any, paramMetadata: ParamMetadata) {
         try {
             const parseValue = typeof value === "string" ? JSON.parse(value) : value;
-            if (paramMetadata.format !== Object && paramMetadata.format && this.driver.useClassTransformer) {
-                const options = paramMetadata.classTransformOptions || this.driver.plainToClassTransformOptions;
-                return plainToClass(paramMetadata.format, parseValue, options);
+            if (paramMetadata.format !== Object && paramMetadata.format) {
+                if (this.driver.useParamValidator) {
+                    const options = paramMetadata.paramValidatorOptions || this.driver.paramValidatorOptions;
+                    return transformAndValidate(paramMetadata.format, parseValue, options)
+                        .catch(err => {
+                            const error: any = new BadRequestError(`Invalid ${paramMetadata.type}, check 'errors' property for more info`);
+                            error.errors = err;
```

Comment:
`transformAndValidate()` throws `ValidationError[]` so this array can't be just returned as JSON and you can't just add the message and httpCode properties.
I added the `catch` block to transform the `ValidationError[]` to the `BadRequestError` which can be easily handled by the default error handler.
But I will refactor the names inside this block to make it more readable 👍